source/common/network/dns_impl.h

```diff
@@ -1,47 +1,68 @@
 #pragma once
 
+#include "envoy/event/file_event.h"
 #include "envoy/network/dns.h"
 
 #include "common/common/linked_object.h"
 #include "common/event/dispatcher_impl.h"
 
+#include "ares.h"
 #include "event2/event_struct.h"
 
 namespace Network {
 
+class DnsResolverImplPeer;
```

Comment:
I pushed this into a resetChannelForTest() method. Most of this is really very test specific, I don't know if it makes that much sense to put behavior like specifying the server:port CSV into a non-test method. I'm good either way, I can remove the final patch if you prefer.